float8_experimental/float8_linear.py

```diff
@@ -204,8 +218,30 @@ def cast_w_to_float8(
             torch.float8_e4m3fn,
             is_amax_initialized,
         )
+
+        if config.weight_cache_enabled:
+            assert config.allocate_float8_weight_cache_buffers
```

Comment:
Also maybe a better error message, "if you are using weight caching you need to enable allocate_float8_weight_cache_buffers = False before performing module swap or construction of Float8Linear module"